src/object/obj_class.c

```diff
@@ -109,7 +109,8 @@ daos_oclass_name2id(const char *name)
 
 	/* slow search path, it's for tool and not performance sensitive. */
 	for (oc = &daos_obj_classes[0]; oc->oc_id != OC_UNKNOWN; oc++) {
-		if (strncmp(oc->oc_name, name, strlen(name)) == 0)
+		if (strlen(name) == strlen(oc->oc_name) &&
+		    strncmp(oc->oc_name, name, strlen(name)) == 0)
```

Comment:
do you really need strncmp if you just checked the string length?  I think coverity will probably complain about this.  Perhaps it should be strlen(oc->oc_name) == strnlen(name, SOME_MAX_STRLEN) && strcmp(....)

where SOME_MAX_STRLEN is bigger than any of the known names.